### ISSUE TITLE
Dropdown menu a11y checker failures

### ISSUE THREAD

**Issue by k8vance88** [carbon-design-system/carbon#6701]
## Using Dropdown from carbon-components-react has errors in the IBM Equal Access Accessibility Checker tool

> Browser
Chrome

> Automated testing tool and ruleset
latest

## Detailed description

> What version of the Carbon Design System are you using?
 "carbon-components": "10.15.0",
 "carbon-components-react": "7.15.0",

> What WCAG 2.1 checkpoint does the issue violate?
The 'id' "downshift-0-label" specified for the WAI-ARIA property 'aria-labelledby' value is not valid 
Violation: The descendent <div> element with "option" role has no focusable child element
Needs review: The input element does not have an associated visible label 

> Please create a reduced test case in CodeSandbox
>
Easily recreate with 
https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox?file=/src/index.js

with this code:
`import React from 'react';
import { render } from 'react-dom';
import { Button, Dropdown } from 'carbon-components-react';

var items = [{
  id: 'option-0',
  text: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Vitae, aliquam. Blanditiis quia nemo enim voluptatibus quos ducimus porro molestiae nesciunt error cumque quaerat, tempore vero unde eum aperiam eligendi repellendus.'
}, {
  id: 'option-1',
  text: 'Option 1'
}, {
  id: 'option-2',
  text: 'Option 2'
}, {
  id: 'option-3',
  text: 'Option 3'
}, {
  id: 'option-4',
  text: 'Option 4'
}, {
  id: 'option-5',
  text: 'Option 5'
}];

var itemToString = function itemToString(item) {
  return item ? item.text : '';
}

const App = () => (
  <div>
    <Dropdown label='Actions' itemToString={itemToString} items={items}/>
  </div>
);

render(<App />, document.getElementById('root'));
`

## Additional information

- Screenshots or code
![image](https://user-images.githubusercontent.com/29546393/90793400-50364e80-e2d1-11ea-98b6-2a910da2b44e.png)

**Comment by joshblack** [comment]
Hey there @k8vance88! 👋  I believe this should be addressed in the latest version of carbon, or by providing the `titleText` prop so that the label is visible

**Comment by k8vance88** [comment]
Great, I will try it, thanks!

**Comment by k8vance88** [comment]
@joshblack what version is it fixed in?  I tried the latest available in npm:
        "carbon-components": "10.16.0",
        "carbon-components-react": "7.16.0",

It does work to add titleText, but that isn't what we prefer to use.

**Comment by joshblack** [comment]
@k8vance88 good question, let me bump @dakahn. In situations where folks don't want to have a visible label present, is it  possible for these components to have the correct label or is the visible label required?

**Comment by dakahn** [comment]
[WCAG Success Criterion 3.3.2](https://www.w3.org/TR/WCAG21/#labels-or-instructions) requires that form inputs have accessible labeling for both sighted users _and_ users of assistive technologies. For a deeper dive [see the section **Intent** in this Labeling and Instructions guide](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html),

**Comment by tw15egan** [comment]
Would we be able to add a `hideLabel` prop that adds the `bx--visually-hidden` class to the label, but would still keep it there for assisted technology users?

**Comment by dakahn** [comment]
DAP/AC will throw and error if there is a programmatic label (aria) but not a visible label, if I remember correctly.

**Comment by k8vance88** [comment]
Any progress on this?  To claim a11y compliance we will be forced to ship a UI that looks like this:
![image](https://user-images.githubusercontent.com/29546393/95133299-b878ac80-0726-11eb-8049-5e8cf1d45c43.png)
instead of this:
![image](https://user-images.githubusercontent.com/29546393/95133400-e0681000-0726-11eb-83f1-a7eefde67785.png)

**Comment by tw15egan** [comment]
> [WCAG Success Criterion 3.3.2](https://www.w3.org/TR/WCAG21/#labels-or-instructions) requires that form inputs have accessible labeling for both sighted users _and_ users of assistive technologies. For a deeper dive [see the section **Intent** in this Labeling and Instructions guide](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html),

Doesn't seem like there is much wiggle room here, a label is needed to pass a11y tests.

**Comment by dakahn** [comment]
@k8vance88 That screen shot is actually a great example of what the WCAG guidance is trying to have developers maintained. In that second example, once the user makes a selection -- the form element no longer has a visual label. It's replaced by the selection. User's with cognitive disabilities -- or who are returning to a form after some time away could be stuck inferring the form element's role from the options. 

I'm going to close this issue as resolved. Since the first screen shot follows our design guidance.
